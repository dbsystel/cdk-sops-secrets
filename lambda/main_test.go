package main

import (
	"os"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/go-test/deep"
)

func TestMain(t *testing.M) {

	//log.SetOutput(ioutil.Discard)

	v := t.Run()
	// After all tests have run `go-snaps` can check for not used snapshots
	snaps.Clean(t)

	os.Exit(v)
}

func Test_GetS3FileContent(t *testing.T) {
	mocks := getMocks(t)
	data, err := mocks.getS3FileContent(SopsS3File{
		Bucket: "..",
		Key:    "../test-secrets/json/sopsfile.enc-age.json",
	})
	check(err)
	snaps.MatchSnapshot(t, string(data))
}

func Test_UpdateSecret(t *testing.T) {
	mocks := getMocks(t)
	fileName := "4547532a137611d83958d17095c6c2d38ae0036a760c3b79c9dd5957d1c20cf2.yaml"
	inputArn := "arn:${Partition}:secretsmanager:${Region}:${Account}:secret:${SecretId}"
	secretValue := []byte("some-secret-data")

	response, err := mocks.updateSecret(fileName, inputArn, secretValue)
	check(err)

	snaps.MatchSnapshot(t, response)
}
func Test_DecryptSopsFileContent(t *testing.T) {

	os.Setenv("SOPS_AGE_KEY", "AGE-SECRET-KEY-1EFUWJ0G2XJTJFWTAM2DGMA4VCK3R05W58FSMHZP3MZQ0ZTAQEAFQC6T7T3")

	sopsEncrypted, err := os.ReadFile("../test-secrets/json/sopsfile.enc-age.json")
	check(err)
	sopsDecrypted, err := decryptSopsFileContent(sopsEncrypted, "json")
	check(err)
	sopsExpected, err := os.ReadFile("../test-secrets/json/sopsfile.json")
	check(err)

	sopsDecryptedJ := UnmarshalAny(sopsDecrypted)
	sopsExpectedJ := UnmarshalAny(sopsExpected)

	if diff := deep.Equal(sopsDecryptedJ, sopsExpectedJ); diff != nil {
		t.Error(diff)
	}
}
