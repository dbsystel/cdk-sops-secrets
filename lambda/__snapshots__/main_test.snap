
[Test_GetS3FileContent - 1]
>>>S3MAnagerMockClient.Download.Input
{
  Bucket: "..",
  Key: "../test-secrets/json/sopsfile.enc-age.json"
}
---

[Test_GetS3FileContent - 2]
{
    "key1": "ENC[AES256_GCM,data:ArSjC0Vd,iv:NFrIj1ekM2dU8JCU6NpmnHFmwcFaEQagCw/fxKV2b6A=,tag:GrGgeNmtqFuyiTjKJ/78kw==,type:str]",
    "key2": "ENC[AES256_GCM,data:NeoandY=,iv:2oCcBk0R/XxvH/O+aaCIa7V9pFsy7dhCKZ70ebjc5Ds=,tag:GVZXUFFwbpMVY/KCLNENAQ==,type:float]",
    "key3": "ENC[AES256_GCM,data:y4tBaNg=,iv:mHuYrtf0Ke/oHBcunpgdPpzaYkBx7UjMcOQgpmVc8/Q=,tag:cuEg2UE17Po6opfpH8ikog==,type:bool]",
    "sops": {
        "kms": null,
        "gcp_kms": null,
        "azure_kv": null,
        "hc_vault": null,
        "age": [
            {
                "recipient": "age1djllw2pzuprrqc0en5m8vc8k5ge3tm0f6g7cj0c0glfzp44vdc4ql8ngvu",
                "enc": "-----BEGIN AGE ENCRYPTED FILE-----\nYWdlLWVuY3J5cHRpb24ub3JnL3YxCi0+IFgyNTUxOSBOWlAxekVNNXdjenV5cmdR\nZDlQU3N5WC9oeENUbUpYcWxkeU9kVS9TeUR3Cm5zbGhIL0J6WG9TWi9KR3ZUTVph\nblpiUDZtQklnWlRrSGZzTXlZN2Z6WjgKLS0tIDI3ZHNHR0NmWTMzdTg0dEtXcU93\nbEdNWndKdWROWGdLdHAwTGFtRGlsaDQKV7HJG629rPWDBY046Hxj4utxUkex3SwU\nVUQRX00p6r9ffc+iC5DGUm/KOketAHunO4Kn0uOS4WHg+Jg2Cwu72Q==\n-----END AGE ENCRYPTED FILE-----\n"
            }
        ],
        "lastmodified": "2022-03-29T21:02:32Z",
        "mac": "ENC[AES256_GCM,data:2vmzwW74OSJq18GEcScpMH76AWrhJdENlKw2vRzGrQfPixTNcrwJ0neTH+BvGKfR9OlLR2EZTheRphQ0Ikb4SdhfInoDjHo+0jXN7RQ4neTlko4j/YRVek61oePgAKAqcmvL2IhzRGO2VR8nDPuJuvRCmP5wOMOBkerydRKHeDc=,iv:NHWPNWDmuhekJUIJpC1cDJ51lLKTsXLE6dC4tpw/qSI=,tag:yV5ngRdIYZfUgj6ntsY5CQ==,type:str]",
        "pgp": null,
        "unencrypted_suffix": "_unencrypted",
        "version": "3.7.2"
    }
}
---

[Test_UpdateSecret - 1]
>>>SecretsManagerMockClient.PutSecretValue.Input
main.putSecretValueInputNotSecure{
    _:                  struct {}{},
    ClientRequestToken: &"4547532a137611d83958d17095c6c2d38ae0036a760c3b79c9dd5957d1c20cf2.yaml",
    SecretBinary:       nil,
    SecretId:           &"arn:${Partition}:secretsmanager:${Region}:${Account}:secret:${SecretId}",
    SecretString:       &"some-secret-data",
    VersionStages:      nil,
}
---

[Test_UpdateSecret - 2]
{
  ARN: "arn:custom:sopssync:${Region}:${Account}:secret:${SecretId}",
  Name: "ea44991f303ca1fc043cd129e135eb477c9f102e05eff7b06813202bda64638b",
  VersionId: "90543d7645d7aa15732054e8a94dc5409278eb964a5dd9a53faeadb0cd849ca1",
  VersionStages: ["ea44991f303ca1fc043cd129e135eb477c9f102e05eff7b06813202bda64638b"]
}
---

[Test_UpdateSSMParameter - 1]
>>>SecretsManagerMockClient.PutParameterValue.Input
main.putParameterValueInputNotSecure{
    _:              struct {}{},
    AllowedPattern: (*string)(nil),
    DataType:       (*string)(nil),
    Description:    (*string)(nil),
    KeyId:          &"key",
    Name:           &"/foo/bar",
    Overwrite:      &bool(true),
    Policies:       (*string)(nil),
    Tags:           nil,
    Tier:           (*string)(nil),
    Type:           &"SecureString",
    Value:          &"some-secret-data",
}
---

[Test_UpdateSSMParameter - 2]
{
  Tier: "Standard",
  Version: 1
}
---
